app/validation/validator.py

```diff
@@ -506,6 +507,45 @@ def _validate_routing_rule(self, rule, answer_ids_with_group_id, block_or_group)
 
         return errors
 
+    def _validate_answer_routing_value(self, rule, answer_ids_with_parent_id):
+        errors = []
+
+        when_rule = rule.get('goto').get('when')
+        if when_rule:
+            for when_dict in when_rule:
+                amount_matched = 0
+                errors.extend(self._validate_routing_when_dict(when_dict, answer_ids_with_parent_id, amount_matched))
+
+        return errors
+
+    def _validate_routing_when_dict(self, when_dict, answer_ids_with_parent_id, amount_matched):
+        errors = []
+
+        if when_dict.get('value'):
+            rule_value = when_dict.get('value')
+            errors.extend(self._validate_routing_when_dict_value(answer_ids_with_parent_id, amount_matched, rule_value))
+
+        elif when_dict.get('values'):
+            for rule_value in when_dict.get('values'):
+                errors.extend(self._validate_routing_when_dict_value(
+                    answer_ids_with_parent_id, amount_matched, rule_value))
+
+        return errors
+
+    @staticmethod
+    def _validate_routing_when_dict_value(answer_ids_with_parent_id, amount_matched, rule_value):
+        errors = []
+        if isinstance(rule_value, int) is not True:
+            for block in answer_ids_with_parent_id:
+                for answer_block in answer_ids_with_parent_id[block]['answer'].get('options', {}):
+                    if rule_value == answer_block.get('value'):
+                        amount_matched += 1
+            if amount_matched == 0:
+                errors.append(Validator._error_message(
+                    f'Answer option and routing rule values mismatch, missing answer value: '
+                    f'{rule_value}'))
+        return errors
+
```

Comment:
This could be reduced to a method which determines if a single rule value is valid and returns it as a boolean.